### PR TITLE
[Intel Mkl] Adding Mkl + horovod partials and Dockerfiles

### PR DESCRIPTION
**Pull request by claynerobison**
This PR adds support for MPI+Horovod to the Dockerfile partials and generated Dockerfiles. 

MKL support is not explicit because it can be built with the CPU Dockerfiles as they are.

### CODE REVIEW

File: tensorflow/tools/dockerfiles/partials/mpi.partial.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# install libnuma, openssh, wget
+RUN apt-get update && apt-get install -y --no-install-recommends --fix-missing \
+        libnuma-dev \
+        openssh-server \
+        openssh-client \
+        wget && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* || \
+    yum -y update && yum -y install \
```
Comment (angersson) [inline tensorflow/tools/dockerfiles/partials/mpi.partial.Dockerfile]: If this is under Ubuntu, is `yum` going to be available?

**Review by angerson — CHANGES_REQUESTED** [review]
Cool, thanks for working on the Dockerfiles. I have a few short comments -- I'm assuming it's tricky for me to test this myself because I don't have an Mkl machine.

Can you put the new partials in a new directory to emphasize what they're used for -- like `mkl_horovod`, for example?

**Comment by gbaned** [comment]
@claynerobison Could you please address the reviewer comments. Thanks!

**Comment by gbaned** [comment]
@claynerobison Did you get a chance to look on reviewer comments? Please let us know on the update. Thanks!

File: tensorflow/tools/dockerfiles/partials/mpi.partial.Dockerfile (hunk 1 above)
Comment (claynerobison) [reply tensorflow/tools/dockerfiles/partials/mpi.partial.Dockerfile]: This partial is designed to work with both Ubuntu and CentOS. So, the RUN command will work on either base OS. That is why it is not under the Ubuntu folder.

**Comment by claynerobison** [comment]
@angersson You can test these yourself quite easily. Just use any Xeon container/machine on GCP (which is probably all of them!)

**Review by angerson — CHANGES_REQUESTED** [review]
> Can you put the new partials in a new directory to emphasize what they're used for -- like mkl_horovod, for example?

Just this again.

File: tensorflow/tools/dockerfiles/partials/mpi.partial.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            openssh-server \
+            openssh-clients \
+            wget && \
+    yum clean all || \
+    echo "Unsupported Linux distribution. Aborting!" && exit 1
+
+# Install Open MPI
+RUN mkdir /tmp/openmpi && \
+    cd /tmp/openmpi && \
+    wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-4.0.0.tar.gz && \
```
Comment (riverliuintel) [inline tensorflow/tools/dockerfiles/partials/mpi.partial.Dockerfile]: Why horovod is to pull the newest master branch from git repo, but the openmpi is to get a specified revision? Couldn't it get from https://github.com/open-mpi/ompi.git master too?

File: tensorflow/tools/dockerfiles/dockerfiles/mkl/devel-horovod-jupyter.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            openssh-server \
+            openssh-clients \
+            wget && \
+    yum clean all || \
+    echo "Unsupported Linux distribution. Aborting!" && exit 1
+
+# Install Open MPI
+RUN mkdir /tmp/openmpi && \
+    cd /tmp/openmpi && \
+    wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-4.0.0.tar.gz && \
```
Comment (riverliuintel) [inline tensorflow/tools/dockerfiles/dockerfiles/mkl/devel-horovod-jupyter.Dockerfile]: Why horovod is to pull the newest master branch from git repo, but the openmpi is to get a specified revision? Couldn't it get from https://github.com/open-mpi/ompi.git master too?

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by penpornk** [comment]
@gbaned Since @angerson has already reviewed and requested changes, I'd rather not review this too because the PR is rather large. I'm removing my review request.

@claynerobison Could you please address @angerson's change request?

> Can you put the new partials in a new directory to emphasize what they're used for -- like mkl_horovod, for example?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by claynerobison** [comment]
Hi @penpornk, sorry this took so long. I believe I've addressed @angerson request. Please let me know if you'd like any more changes.

**Comment by chuanqi129** [comment]
Hi @angerson, could you help review again since we have create the dir for those Dockerfiles?

**Review by angerson — APPROVED** [review]

**Pull request merged**